packages/core-blockchain/src/blockchain.ts

```diff
@@ -361,18 +365,56 @@ export class Blockchain implements blockchain.IBlockchain {
     /**
      * Process the given block.
      */
-    public async processBlock(block: Interfaces.IBlock, callback): Promise<any> {
-        const result: BlockProcessorResult = await this.blockProcessor.process(block);
+    public async processBlocks(blocks: Interfaces.IBlock[], callback): Promise<any> {
+        const acceptedBlocks: Interfaces.IBlock[] = [];
+        let lastProcessResult: BlockProcessorResult;
+        for (const block of blocks) {
+            lastProcessResult = await this.blockProcessor.process(block);
+
+            if (lastProcessResult === BlockProcessorResult.Accepted) {
+                acceptedBlocks.push(block);
+            }
+        }
 
-        if (result === BlockProcessorResult.Accepted || result === BlockProcessorResult.DiscardedButCanBeBroadcasted) {
+        if (
+            lastProcessResult === BlockProcessorResult.Accepted ||
+            lastProcessResult === BlockProcessorResult.DiscardedButCanBeBroadcasted
+        ) {
             // broadcast only current block
-            const blocktime: number = config.getMilestone(block.data.height).blocktime;
+            const blocktime: number = config.getMilestone(blocks[blocks.length - 1].data.height).blocktime;
```

Comment:
Store `blocks[blocks.length - 1]` in a variable named `currentBlock` so it is more clear what the value is.